### PR TITLE
Responsive invoice list

### PR DESCRIPTION
**Pull request by vokimon**
## Description

Invoices page switches from table to list in smaller screen formats.

Screenshot_20240220-162338

## Changes

- New component ListEditor displaying a list of objects using a list widget
- New component MultiDataEditor switching between ListEditor and TableEditor depending on the screen width
- Invoice Pages uses MultiDataEditor as basis

## Observations

- TODO: List editor still not feature complete with TableEditor. Pending:
    - No selection implemented
    - Actions other than item action  (missing: global actions, selection actions, click action)
    - Display is not that smart (it does not uses column info to infer the display)
- TODO: Move ListEditor and MultiDataEditor to components and later to somenergia-ui

## Please, review

- which information is displayed and which is not
- the chosen item layout
- labels (or the lack of it) for each attribute
- icons for the concepts (market...)
- whether the changes on payment status display have sense
- interfaces of TableEditor, ListEditor, MultiDataEditor have sense and are analogous
- naming for the generic components and their attributes...
- it works properly in different sizes

## How to check the new features

- Mount the ov as usual, and use the mobile simulator in the browser (Ctrl+Shift+M in Firefox)

## Deploy notes

- Just the regular deployment

### CODE REVIEW

**Review by javikalsan — COMMENTED** [review]
Design clear and simple, I like the strategy you use to integrate the `ListEditor` using `MultiDataEditor` as a "editor size handler"

About the specifics requested to review:
- No data was lost in the stacked version so, for me, is perfect. The value of 'concept' is represented using text and icon, maybe it's worth leaving the icon alone to avoid information duplication. IMHO the icon gives life/character to the layout, I wouldn't remove it.
- The layout breathes and is well organized
- The icons chosen are quite identifying and fun
- The changes on payment status display have sense since the icons selected to represent the values are quite standard
- The interfaces are sufficiently analogous
- The word `Editor` it confuses me a little bit since we only perform read operations but I'm sure I'm missing context here.
- It works properly in different sizes, I like the 2 small size layouts with and without the sidebar.

File: frontend/src/pages/InvoicesPage/index.jsx [hunk 1]
```diff
@@ -54,6 +122,10 @@ export default function InvoicesPage() {
   const [rows, setRows] = React.useState([])
   const [error, setError] = React.useState(false)
   const { currentUser } = useAuth()
+  const isPaymentEnabled = import.meta.env.VITE_ENABLE_INVOICE_PAYMENT == false
```
[the remark this answers is not in the document]
Comment (vokimon) [reply frontend/src/pages/InvoicesPage/index.jsx:125]: It is a future-"future flag" to enable the credit card icon to pay the invoice. I was using it as component test for having more than one item action besides the pdf download. I am almost sure that domain will ask us to add payment on the future but we could remove it until domain will ask us to add it.

File: frontend/src/pages/InvoicesPage/MultiDataEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import PropTypes from 'prop-types'
```
[the remark this answers is not in the document]
Comment (vokimon) [reply frontend/src/pages/InvoicesPage/MultiDataEditor.jsx:2]: Yep, we should copy them from Table Editor, but i invested too much time in this. Let's do it later.

**Comment by vokimon** [comment]
@javikalsan acknoledged those proposed changes for later since we cannot afford waste more time in this:

- prop-types: do this when we eventually move the component to somenergia-ui
- future-future-flags to pay invoices: maybe cleanup?

Regarding the general comments

- I agree i didn't think too much about components names. "Editor" follows the TableEditor analogy, and the former is Editor because, provided with the proper actions or views, the component enables items edition (think about the PersonEditor in Tomatic) but i agree the name is somewhat weird. Now you have a TableEditor, a ListEditor and a MultiDataEditor (a DataEditor would be a single dictionary editor, so a MultiDataEditor is a list of dictionaries editor). But yes, weird. If you propose a better names for the 3 components i will definitively consider them.

**Pull request merged**